### PR TITLE
Feature/add dehumidifier

### PR DESCRIPTION
**Pull request by dennis1804**

### CODE REVIEW

**Comment by PietPiraat80** [comment]
first of all I want to thank you for the great work so far. As it stands now most functions are available. 

target humidity works fine. Only thing is the value ones selected (slider) will jump back to zero, so it doesn't show the current setting. 

the operations that II can select in home assistant are heat, cool and off. If I select cool, my dehumidifier will turn on. heat does nothing and off is off. So, this needs a bit of tweaking.

the functions (dehumidify, automatic, continuous, laundry, fan) work great. 

the only fan modes that work are low and high. If I look in the app these are the only ones that I can select. the other options in Hass (on, off and average(gemiddeld)) do nothing.

swing mode does nothing. the app only mentions  on or off. 

there is also a function called security (lock and unlock) and a timer available in the app. I personally don’t use them but maybe other users will. I will gladly send you the logs so you can add them.

Regards

**Comment by dennis1804** [comment]
Thanks, so target humidity needs some tweaking. 
Is there a heat mode in the homewizard app? Or would it be better to remove it from this plugin?

I'll remove the other fan modes that do nothing and see what's wrong with the swing mode, maybe horizontal isn't the right value.

I'm not sure about the timer, because it's a pretty basic feature in the app, and home assistant has some better implementations like setting up an automation.

Security I will take a look at.

Thanks again for the testing :)

**Comment by PietPiraat80** [comment]
there is no heat mode on the dehumidifier, just on and off.

swing mode only has on or off as a possibility.

one thing that is missing is the current humidity. the app shows both current temperature (which is working) and current humidity.

**Comment by PietPiraat80** [comment]
just looked at the debug log in Hass and it does show the changes to current humidity. It just doesn’t show on the card, probably need to try another.

`2023-06-10 08:44:34.084 DEBUG (Thread-180 (connect)) [custom_components.homewizard_climate.climate.dehumidifier/3c39e721f828] State updated, diff: power_on: False -> True, 
2023-06-10 08:44:48.712 DEBUG (Thread-180 (connect)) [custom_components.homewizard_climate.climate.dehumidifier/3c39e721f828] State updated, diff: target_humidity: 55 -> 50, 
2023-06-10 08:45:01.907 DEBUG (Thread-180 (connect)) [custom_components.homewizard_climate.climate.dehumidifier/3c39e721f828] State updated, diff: fan_speed: high -> low, 
2023-06-10 08:45:05.543 DEBUG (Thread-180 (connect)) [custom_components.homewizard_climate.climate.dehumidifier/3c39e721f828] State updated, diff: current_humidity: 63 -> 62, 
2023-06-10 08:45:09.202 DEBUG (Thread-180 (connect)) [custom_components.homewizard_climate.climate.dehumidifier/3c39e721f828] State updated, diff: fan_speed: low -> high, 
2023-06-10 08:45:21.997 DEBUG (Thread-180 (connect)) [custom_components.homewizard_climate.climate.dehumidifier/3c39e721f828] State updated, diff: mode: dehumidify -> fan, 
2023-06-10 08:45:28.780 DEBUG (Thread-180 (connect)) [custom_components.homewizard_climate.climate.dehumidifier/3c39e721f828] State updated, diff: fan_speed: high -> low, 
2023-06-10 08:45:48.518 DEBUG (Thread-180 (connect)) [custom_components.homewizard_climate.climate.dehumidifier/3c39e721f828] State updated, diff: power_on: True -> False, 
2023-06-10 08:46:16.812 DEBUG (Thread-180 (connect)) [custom_components.homewizard_climate.climate.dehumidifier/3c39e721f828] State updated, diff: current_humidity: 62 -> 61, 
2023-06-10 08:48:48.975 DEBUG (Thread-180 (connect)) [custom_components.homewizard_climate.climate.dehumidifier/3c39e721f828] State updated, diff: current_humidity: 61 -> 62, 
2023-06-10 08:50:36.480 DEBUG (Thread-180 (connect)) [custom_components.homewizard_climate.climate.dehumidifier/3c39e721f828] State updated, diff: current_humidity: 62 -> 59, 
2023-06-10 08:50:36.687 DEBUG (Thread-180 (connect)) [custom_components.homewizard_climate.climate.dehumidifier/3c39e721f828] State updated, diff: current_temperature: 17 -> 18, 
2023-06-10 08:50:37.693 DEBUG (Thread-180 (connect)) [custom_components.homewizard_climate.climate.dehumidifier/3c39e721f828] State updated, diff: current_humidity: 59 -> 62, 
2023-06-10 08:50:38.009 DEBUG (Thread-180 (connect)) [custom_components.homewizard_climate.climate.dehumidifier/3c39e721f828] State updated, diff: current_temperature: 18 -> 17, 
2023-06-10 08:51:15.362 DEBUG (Thread-180 (connect)) [custom_components.homewizard_climate.climate.dehumidifier/3c39e721f828] State updated, diff: power_on: False -> True, 
2023-06-10 08:51:21.231 DEBUG (Thread-180 (connect)) [custom_components.homewizard_climate.climate.dehumidifier/3c39e721f828] State updated, diff: current_humidity: 62 -> 59, 
2023-06-10 08:51:21.372 DEBUG (Thread-180 (connect)) [custom_components.homewizard_climate.climate.dehumidifier/3c39e721f828] State updated, diff: current_temperature: 17 -> 18, 
2023-06-10 08:51:22.423 DEBUG (Thread-180 (connect)) [custom_components.homewizard_climate.climate.dehumidifier/3c39e721f828] State updated, diff: current_humidity: 59 -> 62, 
2023-06-10 08:51:22.631 DEBUG (Thread-180 (connect)) [custom_components.homewizard_climate.climate.dehumidifier/3c39e721f828] State updated, diff: current_temperature: 18 -> 17, 
2023-06-10 08:51:38.202 DEBUG (Thread-180 (connect)) [custom_components.homewizard_climate.climate.dehumidifier/3c39e721f828] State updated, diff: current_humidity: 62 -> 61, 
2023-06-10 08:52:18.268 DEBUG (Thread-180 (connect)) [custom_components.homewizard_climate.climate.dehumidifier/3c39e721f828] State updated, diff: swing: True -> False, 
2023-06-10 08:52:19.195 DEBUG (Thread-180 (connect)) [custom_components.homewizard_climate.climate.dehumidifier/3c39e721f828] State updated, diff: swing: False -> True, 
2023-06-10 08:53:15.015 DEBUG (Thread-180 (connect)) [custom_components.homewizard_climate.climate.dehumidifier/3c39e721f828] State updated, diff: power_on: True -> False, 
2023-06-10 08:55:31.586 DEBUG (Thread-180 (connect)) [custom_components.homewizard_climate.climate.dehumidifier/3c39e721f828] State updated, diff: current_humidity: 61 -> 62, 
2023-06-10 08:55:37.627 DEBUG (Thread-180 (connect)) [custom_components.homewizard_climate.climate.dehumidifier/3c39e721f828] State updated, diff: current_humidity: 62 -> 59, 
2023-06-10 08:55:37.769 DEBUG (Thread-180 (connect)) [custom_components.homewizard_climate.climate.dehumidifier/3c39e721f828] State updated, diff: current_temperature: 17 -> 18, 
2023-06-10 08:55:38.734 DEBUG (Thread-180 (connect)) [custom_components.homewizard_climate.climate.dehumidifier/3c39e721f828] State updated, diff: current_humidity: 59 -> 62, 
2023-06-10 08:55:38.986 DEBUG (Thread-180 (connect)) [custom_components.homewizard_climate.climate.dehumidifier/3c39e721f828] State updated, diff: current_temperature: 18 -> 17, 
2023-06-10 08:55:40.035 DEBUG (Thread-180 (connect)) [custom_components.homewizard_climate.climate.dehumidifier/3c39e721f828] State updated, diff: current_humidity: 62 -> 59, 
2023-06-10 08:55:40.266 DEBUG (Thread-180 (connect)) [custom_components.homewizard_climate.climate.dehumidifier/3c39e721f828] State updated, diff: current_temperature: 17 -> 18, 
2023-06-10 08:55:41.251 DEBUG (Thread-180 (connect)) [custom_components.homewizard_climate.climate.dehumidifier/3c39e721f828] State updated, diff: current_humidity: 59 -> 62, 
2023-06-10 08:55:41.542 DEBUG (Thread-180 (connect)) [custom_components.homewizard_climate.climate.dehumidifier/3c39e721f828] State updated, diff: current_temperature: 18 -> 17, 
2023-06-10 08:55:44.774 DEBUG (Thread-180 (connect)) [custom_components.homewizard_climate.climate.dehumidifier/3c39e721f828] State updated, diff: current_temperature: 17 -> 18, 
2023-06-10 08:55:45.010 DEBUG (Thread-180 (connect)) [custom_components.homewizard_climate.climate.dehumidifier/3c39e721f828] State updated, diff: current_humidity: 62 -> 59, 
2023-06-10 08:55:47.320 DEBUG (Thread-180 (connect)) [custom_components.homewizard_climate.climate.dehumidifier/3c39e721f828] State updated, diff: current_humidity: 59 -> 62, 
2023-06-10 08:55:47.523 DEBUG (Thread-180 (connect)) [custom_components.homewizard_climate.climate.dehumidifier/3c39e721f828] State updated, diff: current_temperature: 18 -> 17, 
2023-06-10 08:55:49.784 DEBUG (Thread-180 (connect)) [custom_components.homewizard_climate.climate.dehumidifier/3c39e721f828] State updated, diff: current_humidity: 62 -> 59, 
2023-06-10 08:55:50.075 DEBUG (Thread-180 (connect)) [custom_components.homewizard_climate.climate.dehumidifier/3c39e721f828] State updated, diff: current_temperature: 17 -> 18, 
2023-06-10 08:56:47.828 DEBUG (Thread-180 (connect)) [custom_components.homewizard_climate.climate.dehumidifier/3c39e721f828] State updated, diff: current_humidity: 59 -> 60, `

**Comment by dennis1804** [comment]
hi @PietPiraat80, i updated the swing, fan and humidity settings, can you check them? couldn't find a suitable entry for the lock so i might fix that later :)

thanks in advance,

**Comment by PietPiraat80** [comment]
First of all, the good news is that the current humidity is displayed, and the target humidity shows the set value. The extra functions that were present but didn't work are now gone. Now it only displays "cool" (which acts as "on") and "off". Is it possible to change this to "on" and "off"?

The fan modes, low and high, are shown and function as intended. The only issue is that once the value is set, it disappears, leaving the field blank.

The swing mode doesn't do anything. It's always on, and when I change it to "off", it flips back to "on" within 2 seconds. Nothing changes in the dehumidifier native app.

**Comment by PietPiraat80** [comment]
I've run the debug script again so yo can see the values used, maybe this will help fix the last issues. 

[code.txt](https://github.com/dennis1804/homewizard-climate-hass/files/11886564/code.txt)

**Comment by dennis1804** [comment]
so i might have fixed the fan speed and swing. could you check that out.
there is no "on" option for the hvac mode (see: https://developers.home-assistant.io/docs/core/entity/climate/#hvac-modes), I think i'll hold back on that one and focus on a little rebuild of the package.
now that we have a couple of different devices and the states that they require i can make a better model for those devices.
that should also allow it to be more flexible with setting different options, like the hvac mode

File: custom_components/homewizard_climate/manifest.json [hunk 1]
```diff
@@ -3,8 +3,8 @@
   "name": "Homewizard Climate",
   "config_flow": true,
   "documentation": "https://www.home-assistant.io/integrations/homewizard_climate",
-  "requirements": ["homewizard-climate-ws==0.0.18"],
+  "requirements": ["homewizard-climate-ws==0.0.22"],
```
Comment (dennis1804) [inline custom_components/homewizard_climate/manifest.json:6]:
```suggestion
  "requirements": ["homewizard-climate-ws==0.0.23"],
```

**Pull request merged**